client/json-rpc/src/client.rs

```diff
@@ -261,12 +262,29 @@ impl JsonRpcAsyncClient {
             )));
         }
 
-        let responses: Vec<Value> = resp
+        let json: Value = resp
             .json()
             .await
             .map_err(|e| JsonRpcAsyncClientError::InvalidServerResponse(e.to_string()))?;
-        process_batch_response(batch, responses)
-            .map_err(|e| JsonRpcAsyncClientError::InvalidServerResponse(e.to_string()))
+        match json {
```

Comment:
I think you can write it like this

```
if let Value::Array(response) = json {
  process_batch_response(batch, responses)
                .map_err(|e| JsonRpcAsyncClientError::InvalidServerResponse(e.to_string()))
} else if let Value::Object(response) = json {
   ...
} 
  Err(JsonRpcAsyncClientError::InvalidServerResponse(
                    "expect batch response array instead of object".to_string(),
                ))

```